src/main/java/org/mockito/internal/invocation/TypeSafeMatching.java

```diff
@@ -65,6 +65,6 @@ private static boolean isMatchesMethod(Method method) {
         if (method.isBridge()) {
             return false;
         }
-        return method.getName().equals("matches");
+        return "matches".equals(method.getName());
```

Comment:
@TimvdLippe in this case the flip add no value since the Method.getName() never returns null, it makes the code less readable